lib/vinyl-stream.js

```diff
@@ -35,6 +35,9 @@ module.exports = function createVinylStream(options) {
 				path: base + '/' + object.Key
 			});
 
+	  if(!file.contents)
+		  file.contents =  new Buffer("");
+
 		// Set the length on things like streams which are used
 		// by some frameworks.
 		if (!file.contents.length) {
```

Comment:
The file won't be null, and if it was, it'd defeat the point of getting file metadata.  The only thing that is null is the file contents.

And really, if you're calling it such that the file contents are empty (which means that you're choosing not to read the file contents, as in the case of your "meta" example), then you may want to set the content length so that you can make use of the file size.

In order to do that, you need to make the contents into a buffer, or Vinyl will throw exceptions.